export/export_pipeline/compute_collision_representation.py

```diff
@@ -0,0 +1,427 @@
+                
+                # merge the geoms
+                for l in geom_group[1:]:
+                    l[1].select_set(True)
+                    shutil.rmtree(l[0])
+                geom.select_set(True)
+                bpy.context.view_layer.objects.active = geom
+                bpy.ops.object.join()
+                
+                # Calculate the number of faces in the mesh
+                poly_count = len(geom.data.polygons)
+                print("Polygon count: ", poly_count)
+                
+                # Get the adaptive decimation ratio
+                decimation_ratio = decimation_lookup(poly_count)
+                    
+                # Add the Decimate modifier to the selected object
+                print("Applying decimation with a ratio: ", decimation_ratio)
+                decimate_modifier = geom.modifiers.new(name='Decimate', type='DECIMATE')
+                decimate_modifier.ratio = decimation_ratio
+
+                # Exporting in stl.
+                stl_filepath = osp.join(geom_dir, f"{leaf_name}.stl")
+                if (not self.rerun) or (not osp.isfile(stl_filepath)):
+                    print("Exporting STL file...")
+                    bpy.ops.export_mesh.stl(
+                        filepath=stl_filepath,
+                        check_existing=False,
+                        use_selection=True,
+                        use_mesh_modifiers=True,
+                        batch_mode='OFF',
+                        global_scale=1.0,
+                        axis_forward='Y',
+                        axis_up='Z',
+                        ascii=False,
+                    )
+                bpy.ops.object.select_all(action='DESELECT')
+
+                params = copy.deepcopy(self.p)
+                params['collision_representation']['acd']['decompose_method'] = decompose_method
+                process_mesh_args.append((stl_filepath, params, self.rerun))
+
+        print("Convex solid decomposition...")
+        i = 0
+        while True:
+            start_idx = i * self.n_workers
+            end_idx   = (i+1) * self.n_workers
+            with mp.Pool(self.n_workers) as p:
+                p.map(process_mesh, process_mesh_args[start_idx : end_idx])
+            i = i + 1
+            if end_idx >= len(process_mesh_args):
+                break
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--scene_path', required=True, help='relative path to the scene folder')
+    parser.add_argument('--rerun', action='store_true', help='re-run the export without running decomposition')
+    parser.add_argument('--objects', help='comma separated', default=None)
+    parser.add_argument('-n', type=int, help='number of parallel workers to use', default=mp.cpu_count()-1)
+    args = parser.parse_args()
+    
+    crc = CollisionRepresentationComputer(osp.expanduser(args.scene_path), args.n, args.rerun, args.objects)
```

Comment:
I think it would be a good idea to have a high-level function that acts like a table of contents and looks like this:

```
for actor_gltf in actor_gltfs:

    # compute a component desc for each component in the current per-actor GLTF
    merged_static_mesh_component_descs, physics_constraint_component_descs = compute_component_descs(...)

    # save component descs for downstream processing
    export_component_descs(merged_static_mesh_components, physics_constraint_components)

    # decimate each set of merged StaticMeshComponents, save the decimated results for downstream processing
    decimate_and_export_merged_static_mesh_components(merged_static_mesh_components)

    # compute convex decomposition for each set of merged StaticMeshComponents
    if args.use_single_threaded_reference_implementation:
        compute_and_export_decompositions(merged_static_mesh_components)

# compute convex decomposition for each set of merged StaticMeshComponents
if not args.use_single_threaded_reference_implementation:
    with mp.Pool(...) as pool:
        pool.map(...)
```

Reply:
great idea, done!